partner_identification/models/res_partner.py

```diff
@@ -147,10 +157,11 @@ def _search_identification(self, category_code, operator, value):
         Returns:
             list: Domain to search with.
         """
-        id_numbers = self.env['res.partner.id_number'].search([
-            ('name', operator, value),
-            ('category_id.code', '=', category_code),
-        ])
-        return [
-            ('id_numbers.id', 'in', id_numbers.ids),
-        ]
+        result = []
+        for arg in args:
+            if isinstance(arg, tuple) and arg[0] == field_name:
+                result.append(
+                    ('id_numbers.name',
+                     arg[1],
+                     arg[2]))
```

Comment:
this will yield records that have some identification matching the input. I think you need to mimic what the above does, possibly inspecting self._fields for the category code

Reply:
The problem is that the search method does not get an extra argument, like the compute function and the reverse function do. There is no one to one correspondence between the field name and the category. On the other hand, ID numbers tend to have a quite different numbering scheme for different categories. So mostly it will be extremely unlikely that you will find some person by passport number, searching with the number from the drivers license.

Reply:
that's why I propose to look at `self._fields[field_name]`, there you'll find the `arg` passed to the field